cpg-core/src/main/kotlin/de/fraunhofer/aisec/cpg/graph/Extensions.kt

```diff
@@ -698,52 +719,82 @@ inline fun Node.followXUntilHit(
     val fulfilledPaths = mutableListOf<List<Node>>()
     // failedPaths: All the paths which do not satisfy "predicate"
     val failedPaths = mutableListOf<List<Node>>()
+    val loopingPaths = mutableListOf<List<Node>>()
     // The list of paths where we're not done yet.
-    val worklist = mutableSetOf<Pair<List<Node>, Context>>()
-    worklist.add(Pair(listOf(this), context)) // We start only with the "from" node (=this)
+    val worklist = mutableSetOf<List<Pair<Node, Context>>>()
```

Comment:
We changed this to an identity set in #2143. Maybe other changes are also not transferred?

Reply:
Changed to identity set